tracer/src/Datadog.Trace/Util/IdGenerator.cs

```diff
@@ -0,0 +1,36 @@
+// <copyright file="IdGenerator.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+namespace Datadog.Trace.Util;
+
+/// <summary>
+/// Generates random numbers suitable for use as Datadog trace ids and span ids.
+/// </summary>
+internal class IdGenerator
+{
+    /// <summary>
+    /// Returns a random number greater than zero and less than Int64.MaxValue.
+    /// </summary>
+    public static ulong NextUInt64()
```

Comment:
So you're saying this _isn't_ returning a `UInt64`, it's returning `UInt63`, right? 🤔 I know we currently use 63-bit for our span IDs, but I wonder if the name should make that clear?

Reply:
Plus, it returns a `ulong`, which is by definitely a `UInt64`, regardless of the contents of its bits.

Reply:
> Plus, the method returns a ulong, which is by definition a UInt64, regardless of the contents of its bits.

Yes, true.

> System.Random uses a scheme where the minimum is inclusive and the maximum is exclusive, and it defaults to positive integers only, so I think it's standard?

Ok, but as this method returns `[0, Int64.MaxValue)` - 63 random bits (-1 excluding 0). Based on that, it should be named `NextInt64()`, right? Otherwise I would expect it to return `[0, UInt64.MaxValue)`?

Reply:
Oh no, now you've done it! Maybe we should call it `NextSpanId()`, since soon we will need a `NextTraceId()` which returns 128 random bits. This class is not meant to be a general-purpose PRNG anyway.

Reply:
😂 Maybe that's safest 😛

Reply:
I've made a few changes to make things better (I hope). This type is now called `RandomIdGenerator` and the method is `NextSpanId()`.